crates/contract/tests/sandbox/vote.rs

```diff
@@ -853,3 +855,63 @@ async fn test_successful_resharing_after_cancellation_clears_cancelled_epoch_id(
 
     Ok(())
 }
+
+#[tokio::test]
+async fn vote_new_parameters_errors_if_new_participant_is_missing_valid_attestation() {
+    let (worker, contract, mut current_participant_accounts, _) = init_env_secp256k1(1).await;
+
+    let state: ProtocolContractState = contract.view("state").await.unwrap().json().unwrap();
+    let ProtocolContractState::Running(initial_running_state) = state else {
+        panic!("State is not running: {:#?}", state)
+    };
+
+    let initial_epoch_id = initial_running_state.keyset.epoch_id;
+    let existing_params = initial_running_state.parameters.clone();
+    let mut new_participants = existing_params.participants().clone();
```

Comment:
Isn’t `new_participants` a bit confusing in this context? It’s actually the old participants plus one new participant, right?